vllm/model_executor/layers/sampler.py

```diff
@@ -475,32 +484,69 @@ def _multinomial(
     seq_groups: Optional[List[SequenceGroupToSample]] = None,
 ) -> torch.Tensor:
     if num_samples > 1:
-        # This is equivalent to torch.repeat_interleaved (which also
-        # forces a GPU<->CPU sync).
-        # This allows us to do sampling with replacement by creating
-        # num_samples copies of each row in the tensor, and then
-        # batch sampling the resulting tensor.
-        probs = probs[:, None, :].expand(probs.shape[0], num_samples,
-                                         probs.shape[1]).contiguous().view(
-                                             -1, probs.shape[1])
-    q = torch.empty_like(probs)
+        probs = probs.repeat_interleave(num_samples, dim=0)
```

Comment:
According to my experiment `repeat_interleave` do not force GPU synchronization, at least in pytorch 2.3. So changed to use it.